Amplify/Categories/DataStore/Model/Collection/List+Model.swift

```diff
@@ -6,126 +6,136 @@
+
+    public typealias ModelListElement = ModelType
+    public typealias Page = List<ModelType>
     public typealias Element = ModelType
     public typealias Elements = [Element]
 
-    public typealias ArrayLiteralElement = ModelType
-
     /// The array of `Element` that backs the custom collection implementation.
-    internal var elements: Elements
-
-    /// If the list represents an association between two models, the `associatedId` will
-    /// hold the information necessary to query the associated elements (e.g. comments of a post)
-    internal var associatedId: Model.Identifier?
-
-    /// The associatedField represents the field to which the owner of the `List` is linked to.
-    /// For example, if `Post.comments` is associated with `Comment.post` the `List<Comment>`
-    /// of `Post` will have a reference to the `post` field in `Comment`.
-    internal var associatedField: ModelField?
-
-    internal var limit: Int = 100
-
-    /// The current state of lazily loaded list
-    internal var state: LoadState = .pending
+    public var elements: Elements
 
     // MARK: - Initializers
 
-    public convenience init(_ elements: Elements) {
-        self.init(elements, associatedId: nil, associatedField: nil)
-        self.state = .loaded
-    }
-
-    init(_ elements: Elements,
-         associatedId: Model.Identifier? = nil,
-         associatedField: ModelField? = nil) {
+    public init(_ elements: Elements) {
         self.elements = elements
-        self.associatedId = associatedId
-        self.associatedField = associatedField
     }
 
     // MARK: - ExpressibleByArrayLiteral
 
-    required convenience public init(arrayLiteral elements: Element...) {
+    public required convenience init(arrayLiteral elements: ModelType...) {
         self.init(elements)
     }
 
     // MARK: - Collection conformance
 
-    public var startIndex: Index {
-        loadIfNeeded()
-        return elements.startIndex
+    open var startIndex: Int {
+        elements.startIndex
     }
 
-    public var endIndex: Index {
-        return elements.endIndex
+    open var endIndex: Int {
+        elements.endIndex
     }
 
-    public func index(after index: Index) -> Index {
-        return elements.index(after: index)
+    open func index(after index: Index) -> Index {
+        elements.index(after: index)
     }
 
-    public subscript(position: Int) -> Element {
-        return elements[position]
+    open subscript(position: Int) -> Element {
+        elements[position]
     }
 
-    public __consuming func makeIterator() -> IndexingIterator<Elements> {
-        loadIfNeeded()
-        return elements.makeIterator()
+    open __consuming func makeIterator() -> IndexingIterator<Elements> {
+        elements.makeIterator()
     }
 
-    // MARK: - Persistent Operations
+    // MARK: - Persistant operations
 
     public var totalCount: Int {
         // TODO handle total count
         return 0
     }
```

Comment:
self-review; looks like we should mark this as deprecated as well

Reply:
actually this should maintain the same implementation for now, until a sub list implementation supports returning totalCount